tests/pytests/unit/utils/test_network.py

```diff
@@ -1,8 +1,1535 @@
+    ), patch(
+        "salt.utils.network.ip_addrs",
+        MagicMock(return_value=["127.0.0.1", "::1", "fe00::0", "fe02::1"]),
+    ):
+        assert network.generate_minion_id() == "pick.me"
+
+
+def test_generate_minion_id_platform_localhost_addrinfo():
+    """
+    Test if addinfo is picked up.
+
+    :return:
+    """
+    with patch("platform.node", MagicMock(return_value="localhost")), patch(
+        "socket.gethostname", MagicMock(return_value="ip6-loopback")
+    ), patch("socket.getfqdn", MagicMock(return_value="ip6-localhost")), patch(
+        "socket.getaddrinfo",
+        MagicMock(return_value=[(2, 3, 0, "pick.me", ("127.0.1.1", 0))]),
+    ), patch(
+        "salt.utils.files.fopen", mock_open()
+    ), patch(
+        "salt.utils.network.ip_addrs",
+        MagicMock(return_value=["127.0.0.1", "::1", "fe00::0", "fe02::1"]),
+    ):
+        assert network.generate_minion_id() == "pick.me"
+
+
+def test_generate_minion_id_platform_ip_addr_only():
+    """
+    Test if IP address is the only what is used as a Minion ID in case no DNS name.
+
+    :return:
+    """
+    with patch("platform.node", MagicMock(return_value="localhost")), patch(
+        "socket.gethostname", MagicMock(return_value="ip6-loopback")
+    ), patch("socket.getfqdn", MagicMock(return_value="ip6-localhost")), patch(
+        "socket.getaddrinfo",
+        MagicMock(return_value=[(2, 3, 0, "localhost", ("127.0.1.1", 0))]),
+    ), patch(
+        "salt.utils.files.fopen", mock_open()
+    ), patch(
+        "salt.utils.network.ip_addrs",
+        MagicMock(return_value=["127.0.0.1", "::1", "fe00::0", "fe02::1", "1.2.3.4"]),
+    ):
+        assert network.generate_minion_id() == "1.2.3.4"
+
+
+def test_gen_mac():
+    with patch("random.randint", return_value=1) as random_mock:
+        assert random_mock.return_value == 1
+        ret = network.gen_mac("00:16:3E")
+        expected_mac = "00:16:3E:01:01:01"
+        assert ret == expected_mac
+
+
+def test_mac_str_to_bytes():
+    pytest.raises(ValueError, network.mac_str_to_bytes, "31337")
+    pytest.raises(ValueError, network.mac_str_to_bytes, "0001020304056")
+    pytest.raises(ValueError, network.mac_str_to_bytes, "00:01:02:03:04:056")
+    pytest.raises(ValueError, network.mac_str_to_bytes, "a0:b0:c0:d0:e0:fg")
+    assert b"\x10\x08\x06\x04\x02\x00" == network.mac_str_to_bytes("100806040200")
+    assert b"\xf8\xe7\xd6\xc5\xb4\xa3" == network.mac_str_to_bytes("f8e7d6c5b4a3")
+
+
+@pytest.mark.slow_test
+def test_generate_minion_id_with_long_hostname():
+    """
+    Validate the fix for:
+
+    https://github.com/saltstack/salt/issues/51160
+    """
+    long_name = "localhost-abcdefghijklmnopqrstuvwxyz-abcdefghijklmnopqrstuvwxyz"
+    with patch("socket.gethostname", MagicMock(return_value=long_name)):
+        # An exception is raised if unicode is passed to socket.getfqdn
+        minion_id = network.generate_minion_id()
+    assert minion_id != "", minion_id
```

Comment:
```suggestion
    assert minion_id != ""
```
Let pytest show the error, the `minion_id` value will be there too

Reply:
done